src/main/java/core/basesyntax/figures/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax.figures;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int DEFAULT_RADIUS = 10;
+    private static final int SIZE_LIMIT = 30;
+    private static final int NUMBER_OF_FIGURES = 5;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+
+        int figureInt = random.nextInt(NUMBER_OF_FIGURES);
+        switch (figureInt) {
+            case 0:
+                return new Circle(random.nextInt(SIZE_LIMIT), colorSupplier.getRandomColor());
+
+            case 1:
+                return new Square(random.nextInt(SIZE_LIMIT), colorSupplier.getRandomColor());
+
+            case 2:
+                return new IsoscelesTrapezoid(random.nextInt(SIZE_LIMIT),
+                        random.nextInt(SIZE_LIMIT),
+                        random.nextInt(SIZE_LIMIT), colorSupplier.getRandomColor());
+
+            case 3:
+                return new RightTriangle(random.nextInt(SIZE_LIMIT),
+                        random.nextInt(SIZE_LIMIT), colorSupplier.getRandomColor());
+
+            case 4:
+                return new Rectangle(random.nextInt(SIZE_LIMIT),
+                        random.nextInt(SIZE_LIMIT), colorSupplier.getRandomColor());
+
+            default:
+                return new Circle(DEFAULT_RADIUS, Color.WHITE.toString());
```

Comment:
It is better to create 5 private methods getRandomCircle(), getRandomSquare(), ... and call them in switch-case construction, this will increase code readability. (Remember private methods should follow after public as helper ones)